lib/block_view.go

```diff
@@ -6575,6 +6614,164 @@ func (bav *UtxoView) GetPostsPaginatedForPublicKeyOrderedByTimestamp(publicKey [
 	return postEntries, nil
 }
 
+func (bav *UtxoView) GetDiamondSendersForPostHash(postHash *BlockHash) (_pkidToDiamondLevel map[PKID]int64, _err error) {
+	handle := bav.Handle
+	dbPrefix := append([]byte{}, _PrefixDiamondedPostHashDiamonderPKIDDiamondLevel...)
+	dbPrefix = append(dbPrefix, postHash[:]...)
+	keysFound, _ := EnumerateKeysForPrefix(handle, dbPrefix)
+
+	// Iterate over all the db keys & values and load them into the view.
+	expectedKeyLength := 1 + HashSizeBytes + btcec.PubKeyBytesLenCompressed + 8
+	for _, key := range keysFound {
+		// Sanity check that this is a reasonable key.
+		if len(key) != expectedKeyLength {
+			return nil, fmt.Errorf("UtxoView.GetDiamondsForPostHash: Invalid key length found: %d", len(key))
+		}
+
+		diamondPostHash := &BlockHash{}
+		copy(diamondPostHash[:], key[1:1+HashSizeBytes])
+		diamondPostEntry := bav.GetPostEntryForPostHash(diamondPostHash)
+		receiverPKIDEntry := bav.GetPKIDForPublicKey(diamondPostEntry.PosterPublicKey)
```

Comment:
this should happen before the for loop is created.  This information is the same for all keys, so we can get a slight bump in performance by only computing these once. we know the post hash prior to the loop creation, so we don't need to get it from the key.

Reply:
resolved with #16